module-2/backend/week_one.md

```diff
@@ -1,32 +1,85 @@
+  ```ruby
+  get '/horses' do
+    @count = Count.first
+    name = "Mr. Ed"
+    erb :index
+  end
+  ```
+
 9. What's the purpose of ERB?
+It's an html file that takes embedded ruby code and logic. Unlike a plain html file, you can 'interpolate' ruby code so that when your view is rendered, the page the the user sees includes the results of the ruby logic.
+
 10. Why do I need a development AND test database?
+With testing, you may be doing things that break or mess up your data.  You could be adding and deleting information or you might do something to totally screw it up. It's important to have a database that is clean for developing, and one that is separate which (if changed) will not affect development.
+
 11. What is CRUD and why is it important?
-12. What does HTTP stand for? 
+CRUD stands for Create, Read, Update, Destroy. It's important because it describes the most basic actions we can perform online. All web content is created using these basic methods.
+
+12. What does HTTP stand for?
+hypertext transfer protocol
+
 13. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+<% %> and <%= %>. The first will not be rendered in the user's screen whereas the second will.
+
 14. What's an ORM?
+Object Relational Map. It's a way of using data from disparate sources by converting each line of a data table into an object. As an object, it's easier to access and perform calculations on methods (the attributes or each cell in the data's row). You can also more easily work with, reference, and perform calculations on data from various tables instead of needing to make one massive unwieldy table.
+
 15. What's the most commonly used ORM in ruby (Sinatra & Rails)?
+Active Record
+
 16. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
-17. What's a migration? 
+* get '/models' - shows user all the instances of the model
+* get '/models/:id' - shows user the data of an individual instance
+* get /models/new - allows user to create a new instance to add to the database. Sends a form, and includes a redirect to another view, usually the /models
+* post '/models' - creates the instance and saves it to the database.
+* get '/models/:id/edit' - presents a form to the user to edit a specific instance.
+* put '/models/:id'- allows user to edit an instance
+* delete '/models/:id'
+
+17. What's a migration?
+I feel like the name doesn't really fit what I think of it as. A migration seems to be a file that sets up the data-type for each column of information that you import from a data source.  When you seed your database, the data is checked against the type that it is supposed to be, and is added to the table as that data type.

```

Comment:
This is fair. I would think of migrations primarily as instructions to make changes to the structure of your database.